src/OffloadManager.php

```diff
@@ -208,6 +208,8 @@ protected function run($key, $task, $options)
 		if ($ttl_fresh || $ttl_stale) {
 			$result->then(function ($data) use ($key, $run, $ttl_fresh, $ttl_stale) {
 				if (!$run->isBad()) {
+					$ttl_fresh = $run->getTtlFresh() ?: $ttl_fresh;
```

Comment:
Maybe this should be a `null` check. Maybe somebody wants to set something to 0 on purpose.